core/prop.h

```diff
@@ -25,14 +25,22 @@ class Property
 {
  public:
   Property(const TransitionSystem & ts, const smt::Term & p);
+
+  /** Copies property to a new solver
+   *  @param prop the property to copy
+   *  @param tt the term translator to use
+   *  @return a property using the solver in tt
+   */
+  Property(const Property & prop, smt::TermTranslator & tt);
+
   ~Property();
 
   const smt::Term prop() const { return prop_; }
 
   const TransitionSystem & transition_system() const { return ts_; }
 
  private:
-  const TransitionSystem & ts_;
+  const TransitionSystem ts_;
```

Comment:
what is the motivation for this change here?

Reply:
Ahh that's because the property constructor allows copying the transition system to a new solver. So the transition system needs to be owned by it. 

Thus, when you create a new engine, it copies the property, which also copies the property's transition system. So the engine actually has a reference to the system and the real owner is the new property object.